demonstrations/tutorial_qutrits_bernstein_vazirani.py

```diff
@@ -0,0 +1,394 @@
+
+  As you can see, we have separated the first three qubits from the fourth for simplicity.
+
+- If we now apply our operator :math:`U_f`,
+
+  .. math::
+      |\phi_2\rangle:= U_f |\phi_1\rangle = \frac{1}{\sqrt{2^3}}\left(\sum_{\vec{z} \in \{0,1\}^3}|\vec{z}\rangle\frac{|\vec{a}\cdot\vec{z} \pmod 2\rangle-|1 + \vec{a}\cdot\vec{z} \pmod 2\rangle}{\sqrt{2}}\right).
+
+  Depending on the value of :math:`f(\vec{x})` the final part of the expression can take two values and it can be checked that
+
+  .. math::
+      |\phi_2\rangle = \frac{1}{\sqrt{2^3}}\left(\sum_{\vec{z} \in \{0,1\}^3}|\vec{z}\rangle(-1)^{\vec{a}\cdot\vec{z}}\frac{|0\rangle-|1\rangle}{\sqrt{2}}\right).
+
+  This is because if :math:`\vec{a}\cdot\vec{z}` takes the value :math:`0`, we will have the :math:`\frac{|0\rangle - |1\rangle}{\sqrt{2}}` and if it takes the value :math:`1`, the result will be :math:`\frac{|1\rangle - |0\rangle}{\sqrt{2}} = - \frac{|0\rangle - |1\rangle}{\sqrt{2}}`. Therefore, by calculating :math:`(-1)^{\vec{a}\cdot\vec{z}}` we cover both cases.
+
+- After this, we can enter the :math:`(-1)^{\vec{a}\cdot\vec{z}}` factor in the :math:`|\vec{z}\rangle` term and disregard the last qubit since we are not going to use it again
+
+  .. math::
+        |\phi_2\rangle =\frac{1}{\sqrt{2^3}}\sum_{\vec{z} \in \{0,1\}^3}(-1)^{\vec{a}\cdot\vec{z}}|\vec{z}\rangle.
+
+  Note that you cannot always disregard a qubit. In cases where there is entanglement with other qubits it is not possible but in this case it is separable.
+
+- Finally, we will reapply the property of the first step to calculate the result after using the Hadamard
+
+  .. math::
+        |\phi_3\rangle := H^3|\phi_2\rangle = \frac{1}{2^3}\sum_{\vec{z} \in \{0,1\}^3}(-1)^{\vec{a}\cdot\vec{z}}\left(\sum_{\vec{y} \in \{0,1\}^3}(-1)^{\vec{z}\cdot\vec{y}}|\vec{y}\rangle\right).
+
+  Rearranging this expression we obtain that
+
+  .. math::
+        |\phi_3\rangle  = \frac{1}{2^3}\sum_{\vec{y} \in \{0,1\}^3}\left(\sum_{\vec{z} \in \{0,1\}^3}(-1)^{\vec{a}\cdot\vec{z}+\vec{y}\cdot\vec{z}}\right)|\vec{y}\rangle.
+
+Perfect! The only thing left to check is that in fact, the previous state is exactly :math:`|\vec{a}\rangle`. It may seem complicated but I invite you to demonstrate it by showing that :math:`\langle \vec{a}|\phi_3\rangle = 1`. Let's go to the code and check that it works.
+
+Algorithm coding with qubits
+------------------------------
+
+We will first code the classical solution. We will do this inside a quantum circuit to understand how to use the oracle but we are really just programming the qubits as bits.
+
+"""
+
+
+import pennylane as qml
+
+dev = qml.device("default.qubit", wires = 4, shots = 1)
+
+def Uf():
+    # The oracle in charge of encoding a hidden "a" value.
+    qml.CNOT(wires = [1,3])
+    qml.CNOT(wires = [2,3])

```

Comment:
```suggestion
    qml.CNOT(wires=[1, 3])
    qml.CNOT(wires=[2 ,3])

```
Recommend to run all the code through black for consistent formatting.